doc/changelog.md

```diff
@@ -1,3 +1,55 @@
+# Release 0.4.1
+
+<h3>Improvements</h3>
+
+* Catalyst wheels are now packaged with OpenMP and ZStd.
+  [(#457)](https://github.com/PennyLaneAI/catalyst/pull/457)
+
+<h3>Bug fixes</h3>
+
+* Resolve an infinite recursion in the decomposition of the `Controlled`
+  operator whenever computing a Unitary matrix for the operator fails.
+  [(#468)](https://github.com/PennyLaneAI/catalyst/pull/468)
+
+* Resolve a failure to generate gradient code for specific input circuits.
+  [(#439)](https://github.com/PennyLaneAI/catalyst/issues/439)
+
+  In this case, [`jnp.mod`](https://github.com/PennyLaneAI/catalyst/issues/437)
+  was used to compute wire values in a for loop, which prevented the gradient
+  architecture from fully separating quantum and classical code. The following
+  program is now supported:
+  ```py
+  @qjit
+  @grad
+  @qml.qnode(dev)
+  def f(x):
+      def cnot_loop(j):
+          qml.CNOT(wires=[j, jnp.mod((j + 1), 4)])
+  
+      for_loop(0, 4, 1)(cnot_loop)()
+  
+      return qml.expval(qml.PauliZ(0))
+  ```
+
+* Resolve unpredictable behaviour when importing libraries that share Catalyst's LLVM dependency
```

Comment:
```suggestion

* Resolve a memory leak in the runtime stemming from  missing calls to device destructors
  at the end of programs.
  [(#446)](https://github.com/PennyLaneAI/catalyst/pull/446)

* Resolve unpredictable behaviour when importing libraries that share Catalyst's LLVM dependency
```

Reply:
@maliasadi I didn't include this PR, how come you are suggesting adding it to the changelog?

Reply:
nwm - I thought that you already included all David's suggestions https://github.com/PennyLaneAI/catalyst/pull/481#issuecomment-1914932651